packages/wallet/primitives/src/network.ts

```diff
@@ -1,630 +1,1229 @@
+      name: 'SEI Explorer',
+      rootUrl: 'https://seitrace.com/?chain=pacific-1',
+    },
+    nativeToken: {
+      symbol: 'SEI',
+      name: 'SEI',
+      decimals: 18,
+    },
   },
-}
 
-export const MonadTestnet: Network = {
-  name: 'monad-testnet',
-  rpc: getRpcUrl('monad-testnet'),
-  chainId: 10143n,
-  explorer: 'https://testnet.monadexplorer.com/',
-  nativeCurrency: {
-    name: 'MON',
-    symbol: 'MON',
-    decimals: 18,
+  {
+    chainId: ChainId.SEI_TESTNET,
+    type: NetworkType.TESTNET,
+    name: 'sei-testnet',
+    title: 'Sei Testnet',
+    rpc: getRpcUrl('sei-testnet'),
+    logoURI: `https://assets.sequence.info/images/networks/medium/${ChainId.SEI_TESTNET}.webp`,
+    blockExplorer: {
+      name: 'Sei Testnet Explorer',
+      rootUrl: 'https://seitrace.com/?chain=atlantic-2',
+    },
+    nativeToken: {
+      symbol: 'SEI',
+      name: 'SEI',
+      decimals: 18,
+    },
   },
-}
 
-export const Somnia: Network = {
-  name: 'somnia',
-  rpc: getRpcUrl('somnia'),
-  chainId: 5031n,
-  explorer: 'https://mainnet.somnia.w3us.site/',
-  nativeCurrency: {
-    name: 'SOMI',
-    symbol: 'SOMI',
-    decimals: 18,
+  {
+    chainId: ChainId.SOMNIA,
+    type: NetworkType.MAINNET,
+    name: 'somnia',
+    title: 'Somnia',
+    rpc: getRpcUrl('somnia'),
+    logoURI: `https://assets.sequence.info/images/networks/medium/${ChainId.SOMNIA}.webp`,
+    blockExplorer: {
+      name: 'Somnia Explorer',
+      rootUrl: 'https://mainnet.somnia.w3us.site/',
+    },
+    nativeToken: {
+      symbol: 'SOMI',
+      name: 'SOMI',
+      decimals: 18,
+    },
   },
+]
+
+function getRpcUrl(networkName: string): string {
+  return `https://nodes.sequence.app/${networkName}`
 }
 
-export const SomniaTestnet: Network = {
-  name: 'somnia-testnet',
-  rpc: getRpcUrl('somnia-testnet'),
-  chainId: 50312n,
-  explorer: 'https://somnia-testnet.socialscan.io/',
-  nativeCurrency: {
-    name: 'STT',
-    symbol: 'STT',
-    decimals: 18,
-  },
+export function getNetworkFromName(networkName: string): Network | undefined {
+  return ALL.find((network) => network.name === networkName)
 }
 
-export const All = [
-  Mainnet,
-  Sepolia,
-  Polygon,
-  PolygonAmoy,
-  PolygonZkEVM,
-  BSC,
-  BSCTestnet,
-  Optimism,
-  OptimismSepolia,
-  Arbitrum,
-  ArbitrumSepolia,
-  ArbitrumNova,
-  Avalanche,
-  AvalancheTestnet,
-  Gnosis,
-  Base,
-  BaseSepolia,
-  Homeverse,
-  HomeverseTestnet,
-  Xai,
-  XaiSepolia,
-  Telos,
-  TelosTestnet,
-  B3,
-  B3Sepolia,
-  ApeChain,
-  ApeChainTestnet,
-  Blast,
-  BlastSepolia,
-  SkaleNebula,
-  SkaleNebulaTestnet,
-  Soneium,
-  SoneiumMinato,
-  ToyTestnet,
-  ImmutableZkEVM,
-  ImmutableZkEVMTestnet,
-  RootNetwork,
-  RootNetworkPorcini,
-  Laos,
-  LaosSigmaTestnet,
-  Moonbeam,
-  MoonbaseAlpha,
-  Etherlink,
-  EtherlinkTestnet,
-  MonadTestnet,
-  Somnia,
-  SomniaTestnet,
-]
+export function getNetworkFromChainId(chainId: ChainId | bigint | number): Network | undefined {
```

Comment:
Should we also support Hex | string here since we could be getting  from a EIP1193 response?